src/time_integration/paired_explicit_runge_kutta/polynomial_optimizer.jl

```diff
@@ -0,0 +1,117 @@
+using LinearAlgebra: eigvals
+using ECOS
+using Convex
+const MOI = Convex.MOI
+
+function filter_eigvals(eig_vals, threshold)
+    filtered_eigvals_counter = 0
+    filtered_eig_vals = Complex{Float64}[]
+
+    for eig_val in eig_vals
+        if abs(eig_val) < threshold
+            filtered_eigvals_counter += 1
+        else
+            push!(filtered_eig_vals, eig_val)
+        end
+    end
+
+    println("$filtered_eigvals_counter eigenvalue(s) are not passed on because they are in magnitude smaller than $threshold \n")
+
+    return length(filtered_eig_vals), filtered_eig_vals
+end
+
+function polynoms(cons_order, num_stage_evals, num_eig_vals,
+                  normalized_powered_eigvals_scaled, pnoms, gamma::Variable)
+    for i in 1:num_eig_vals
+        pnoms[i] = 1.0
+    end
+
+    for k in 1:cons_order
+        for i in 1:num_eig_vals
+            pnoms[i] += normalized_powered_eigvals_scaled[i, k]
+        end
+    end
+
+    for k in (cons_order + 1):num_stage_evals
+        pnoms += gamma[k - cons_order] * normalized_powered_eigvals_scaled[:, k]
+    end
+
+    return maximum(abs(pnoms))
+end
+
+function bisection(cons_order, num_eig_vals, num_stage_evals, dtmax, dteps, eig_vals)
```

Comment:
We should also add a docstring to the Paper by Ketcheson and Ahmadia for reference. (I can do that)